src/cephadm/cephadm

```diff
@@ -3439,6 +3453,186 @@ class CephContainer:
                                 desc=self.entrypoint, timeout=timeout)
         return out
 
+
+#####################################
+
+class MgrListener(Thread):
+    def __init__(self, agent: 'CephadmAgent') -> None:
+        self.agent = agent
+        self.stop = False
+        super(MgrListener, self).__init__(target=self.run)
+
+    def run(self) -> None:
+        listenSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
I'm not sure about a bare socket here. It's not clear where this approach would go as and when the agent functionality extends - assuming the goal is to offload as much work as possible to the hosts to achieve scale.

As it stands you can spam this port, and render the data sent to the endpoint unusable - the data is also in the clear. This is the kind of issue that drove the exporter to adopt https after feedback from the rest of the team.

Reply:
Added 2 way ssl for this socket. Also changed it so it can take on a variable length json string rather than an integer, allowing us to essentially send whatever we want to the agent to make it easy to extend the functionality in the future.